src/main/java/net/imagej/ops/stats/IntegralMean.java

```diff
@@ -0,0 +1,149 @@
+import net.imagej.ops.special.computer.AbstractBinaryComputerOp;
+import net.imglib2.FinalInterval;
+import net.imglib2.Interval;
+import net.imglib2.algorithm.neighborhood.RectangleNeighborhood;
+import net.imglib2.converter.Converter;
+import net.imglib2.converter.RealDoubleConverter;
+import net.imglib2.type.numeric.RealType;
+import net.imglib2.type.numeric.real.DoubleType;
+import net.imglib2.util.Intervals;
+import net.imglib2.view.composite.Composite;
+
+/**
+ * {@link Op} to calculate the {@code stats.mean} from an integral image using a
+ * specialized {@code Cursor} implementation.
+ *
+ * @author Stefan Helfrich (University of Konstanz)
+ * @param <I> input type
+ */
+@Plugin(type = Ops.Stats.IntegralMean.class)
+public class IntegralMean<I extends RealType<I>> extends
+	AbstractBinaryComputerOp<RectangleNeighborhood<Composite<I>>, Interval, DoubleType>
+	implements Ops.Stats.IntegralMean
+{
+
+	@Override
+	public void compute2(RectangleNeighborhood<Composite<I>> input1, Interval input2,
+		DoubleType output)
+	{
+		// computation according to	https://en.wikipedia.org/wiki/Summed_area_table
+		final IntegralCursor<Composite<I>> cursor = new IntegralCursor<>(input1);
+		int dimensions = input1.numDimensions();
+		
+		// Compute \sum (-1)^{dim - ||cornerVector||_{1}} * I(x^{cornerVector})
+		final DoubleType sum = new DoubleType();
+		sum.setZero();
+
+		// Convert from input to return type
+		final Converter<I, DoubleType> conv = new RealDoubleConverter<>();
+		
+		while ( cursor.hasNext() )
+		{
+			final I value = cursor.next().get(0).copy();
+			final DoubleType valueAsDoubleType = new DoubleType();
+			conv.convert(value, valueAsDoubleType);
+			
+			// Obtain the cursor position encoded as corner vector
+			int cornerInteger = cursor.getCornerRepresentation();
+			
+			// Determine if the value has to be added (factor==1) or subtracted (factor==-1)
+			DoubleType factor = new DoubleType(Math.pow(-1.0d, dimensions - IntegralMean.norm(cornerInteger)));
+			valueAsDoubleType.mul(factor);
+			
+			sum.add(valueAsDoubleType);
+		}
+
+		// Compute overlap
+		int area = IntegralMean.overlap(IntegralMean.correctNeighborhoodInterval(input1), input2);
+
+		// Compute mean by dividing the sum divided by the number of elements
+		sum.div(new DoubleType(area));
+
+		output.set(sum);
+	}
+
+	/**
+	 * Compute the overlap between to intervals.
+	 * 
+	 * @param interval1
+	 * @param interval2
+	 * @return area/volume/etc
+	 */
+	public static int overlap(Interval interval1, Interval interval2)
+	{
+		Interval intersection = Intervals.intersect(interval1, interval2);
+		return (int) Intervals.numElements(intersection);
+	}
+
+	/**
+	 * Correct the provided {@link Interval} for the increased neighborhood size
+	 * that is necessary to work with {@link IntegralCursor}.
+	 * 
+	 * @param neighborhood
+	 * @return size-corrected {@link Interval}.
+	 */
+	public static Interval correctNeighborhoodInterval(Interval neighborhood) {	
```

Comment:
There is a method `Intervals.expand` or `Intervals.extend` or something like this in `imglib2`.